lib/cisco_node_utils/vlan.rb

```diff
@@ -199,5 +248,88 @@ def mapped_vni=(vni)
     def default_mapped_vni
       config_get_default('vlan', 'mapped_vni')
     end
+
+    def private_vlan_type
+      config_get('vlan', 'private_vlan_type', id: @vlan_id)
+    end
+
+    def private_vlan_type=(pv_type)
+      Feature.private_vlan_enable
+      fail TypeError unless pv_type && pv_type.is_a?(String)
+      set_args_keys_default
+      if pv_type == default_private_vlan_type
+        pv_type = private_vlan_type
+        @set_args[:vlan] = @vlan_id
+        @set_args[:state] = 'no'
+        @set_args[:type] = pv_type
+        ignore_msg = 'Warning: Private-VLAN CLI removed'
+      else
+        @set_args[:vlan] = @vlan_id
+        @set_args[:state] = ''
+        @set_args[:type] = pv_type
+        ignore_msg = 'Warning: Private-VLAN CLI entered'
+      end
+      result = config_set('vlan', 'private_vlan_type', @set_args)
+      cli_error_check(result, ignore_msg)
+    end
+
+    def default_private_vlan_type
+      config_get_default('vlan', 'private_vlan_type')
+    end
+
+    def private_vlan_association
+      result = config_get('vlan', 'private_vlan_association',
+                          id: @vlan_id)
+      result.gsub(/["\[\]" "]/, '')
+    end
+
+    def private_vlan_association=(config)
```

Comment:
Changed config to vlan_list. 
For the method, since the setter for configuring the new association needs to do remove and add, not sure how I can call just once the config set...I think the way it is should be ok and not too difficult to maintain.